usersimcrs/core/information_need.py

```diff
@@ -0,0 +1,81 @@
+"""Interface to represent an information need.
+
+The information need is divided into two parts: constraints and requests. The
+constraints specify the slot-value pairs that the item of interest must satisfy,
+while the requests specify the slots for which the user wants information.
+"""
+
+from __future__ import annotations
+
+import random
+from collections import defaultdict
+from typing import Any, Dict, List
+
+from usersimcrs.core.simulation_domain import SimulationDomain
+from usersimcrs.items.item_collection import ItemCollection
+
+random.seed(42)
+
+
+def generate_information_need(
```

Comment:
This generates a random information need, if I understand. Would be good to put random in the method name then, and also explain in the docstring how it exactly works. 

More importantly, I don't quite see where this would be used -- do we actually need it? If there is no immediate use for it then I'd not add it yet.

Reply:
I updated the name of the method and docstring description.

This would be used by the user simulator during initialisation, that is before starting a conversation the simulator is given an information need to complete. This information need would be used to customise the replies of the simulator and help during the final evaluation. This part is not implemented yet to avoid a very large PR.